java/ql/lib/ext/javax.servlet.model.yml

```diff
@@ -9,9 +9,15 @@ extensions:
       - ["javax.servlet", "ServletRequest", False, "getParameterNames", "()", "", "ReturnValue", "remote", "manual"]
       - ["javax.servlet", "ServletRequest", False, "getParameterValues", "(String)", "", "ReturnValue", "remote", "manual"]
       - ["javax.servlet", "ServletRequest", False, "getReader", "()", "", "ReturnValue", "remote", "manual"]
-
+      - ["javax.servlet", "ServletRequest", True, "getParameter", "(String)", "", "Parameter[0]", "remote", "ai-manual"]
+      - ["javax.servlet", "ServletRequest", True, "getParameterValues", "(String)", "", "Parameter[0]", "remote", "ai-manual"]
```

Comment:
Is this intended to be `Parameter[0]`? As in, are you trying to model overrides of `ServletRequest.getParameter`? I think this is quite covered by the `ReturnValue` sources of the same callables (which btw should be ordered alphabetically so that we could see the multiple models around the same callable together). But I guess if the custom implementation of `getParameter` contained a sink, this would catch it... Not very likely in my opinion, but 🤷

Reply:
Hm, yes, I agree. As separately discussed with you and @kaeluka, we are not (yet) aiming to catch improperly implemented APIs where the name or documentation implies that they handle unsanitised input safely, but they don't. This is a very interesting class of vulnerabilities to look into, but conceptually somewhat separate from the case where we directly get unsanitised input from a malicious user.